File: demos/jans-chip/ios/Jans-Chip/Models/Networking/Response.swift
//
//  Response.swift
//  Jans-Chip
//
//  Created by Nazar Yavornytskyi on 16.07.2024.
//

import Foundation

public struct Response: Codable, ErrorHandler {
    var errorMessage: String?
    var isSuccess: Bool = true
    
    private enum CodingKeys: String, CodingKey {
        case authenticatorData
        case clientDataJSON
        case signature
    }
    
    var authenticatorData: String?
    var clientDataJSON: String?
    var signature: String?
}

public struct AttestationDataResponse: Codable {
    
    private enum CodingKeys: String, CodingKey {
        case attestationObject
        case clientDataJSON
    }
    
    var attestationObject: String
    var clientDataJSON: String
}

File: demos/jans-chip/android/app/src/main/java/io/jans/chip/model/fido/assertion/result/Response.kt
package io.jans.chip.model.fido.assertion.result

import androidx.room.Ignore

class Response {
    var authenticatorData: String? = null
    // ClientDataJSON clientDataJSON;

    var clientDataJSON: String? = null

    var signature: String? = null

    @Ignore
    var isSuccessful: Boolean? = true

    @Ignore
    var errorMessage: String? = null
}